### PR TITLE
Test running plugins with fake and fresh plugins

### PR DESCRIPTION
**Pull request by azr**
These are mainly acceptance tests, so you can run those local by commenting this for example:

https://github.com/hashicorp/packer/blob/8d8481275f8ff65b733f5f9de121a257c42d4df7/command/init_test.go#L175-L177

### CODE REVIEW

File: command/init_test.go [hunk 1]
```diff
@@ -52,19 +71,27 @@ func TestInitCommand_Run(t *testing.T) {
...
-			packer {
-				required_plugins {
-					comment = {
-						source  = "github.com/sylviamoss/comment"
-						version = "v0.2.018"
-					}
-				}
-			}`,
-			cfg.dir("1"),
+			map[string]string{
```
Comment (nywilken) [inline command/init_test.go:74]: Would it be possible to test that trying to install a plugin that is not a multi-plugin fails?

**Review by nywilken — APPROVED** [review]
This looks good to me. Nicely done.

File: command/init_test.go (hunk 1 above)
Comment (nywilken) [reply command/init_test.go:74]: Sorry I realized I just asked a question with no context. 

Re: testing the failing case. I'm curious if it is something we could test for and if it makes sense to include in this PR. I know that we document that only muli-plugins work with `packer init` but I think that, if possible, a test case could help illustrate the case for folks reading the source code.

Comment (azr) [reply command/init_test.go:74]: Sure ! Good idea ! I'll open a new PR.

**Pull request merged**